doc/releases/migration-guide-4.1.rst

```diff
@@ -170,6 +170,16 @@ Entropy
 GNSS
 ====
 
+GPIO
+====
+
+* Renamed the device tree property ``pin_mask`` to ``pin-mask``.
+* Renamed the device tree property ``pinmux_mask`` to ``pinmux-mask``.
+* Renamed the device tree property ``vbatts_pins`` to ``vbatts-pins``.
+* Renamed the device tree property ``bit_per_gpio`` to ``bit-per-gpio``.
+* Renamed the device tree property ``off_val`` to ``off-val``.
+* Renamed the device tree property ``on_val`` to ``on-val``.
```

Comment:
`s/device tree/Devicetree/g` but this can be addressed at a later stage when cleaning up release documentation closer to release date.